src/main/scala/com/fulcrumgenomics/commons/CommonsDef.scala

```diff
@@ -232,6 +232,24 @@ class CommonsDef {
     }
   }
 
+  /**
+    * Implicit that provides additional methods to sum the values of a sequence after applying a mapping to the items.
+    * This has the benefit of not creating any intermediate collections or iterators.   The mapping must map to a
+    * [[Numeric]] type.
+    * @param sequence the sequence of items to transform and sum.
+    * @tparam A the type of items to transform.
+    */
+  implicit class FoldSum[A](sequence: TraversableOnce[A]) {
+    /**
+      * Applies the given transform to the items in sequence, and then sums them.
+      * @param f the transform to map items from type [[A]] to type [[B]].
+      * @param x the numeric type of [[B]].
+      * @tparam B the type of the items after the transform has been applied; must be a [[Numeric]] type.
+      * @return
+      */
+    def foldSum[B](f: A => B)(implicit x: Numeric[B]): B = sequence.foldLeft(x.zero) { case (left, right) => x.plus(left, f(right)) }
```

Comment:
I like this, except how about calling it `sumBy`?  I think that's a little more in line with `maxBy` and `minBy` which exist in the standard API and do a mathematical operation over some transform of a sequence.